testing/unit_tests/test_loader.py

```diff
@@ -0,0 +1,55 @@
+import os
+import subprocess
+
+import pytest
+import pandas as pd
+
+from ivadomed.loader import utils as imed_loader_utils
+
+@pytest.mark.parametrize('loader_parameters', [{
+    "bids_path": "testing_data/microscopy_png",
+    "bids_config": "ivadomed/config/config_bids.json",
+    "target_suffix": [["_seg-myelin-manual", "_seg-axon-manual"]],
+    "extensions": [".png"],
+    "contrast_params": {
+        "training_validation": [],
+        "testing": [],
+        "balance": {}
+    }}])
+def test_bids_df_microscopy_png(loader_parameters):
+    # Test for microscopy png file format
+    # Test for _sessions.tsv and _scans.tsv files
+    # Test for target_suffix as a nested list
+    # Test for when no contrast_params are provided
+    loader_params = loader_parameters
+    loader_params["contrast_params"]["contrast_lst"] = loader_params["contrast_params"]["training_validation"]
+    derivatives = True
+    df = imed_loader_utils.create_bids_dataframe(loader_params, derivatives)
+    df = df.drop(columns=['path', 'parent_path'])
+    df = df.sort_values(by=['filename']).reset_index(drop=True)
+    df.to_csv("testing_data/microscopy_png/df_test.csv", index=False)
+    command = "diff testing_data/microscopy_png/df_test.csv testing_data/microscopy_png/df_ref.csv"
```

Comment:
Here, we run the diff of both files, but it doesn't test if they are the same, which I believe is the goal of this test?

Reply:
The diff command exit code are 0 for identical and 1 for different, so:
If they are the same, the test passes (exit code 0).
If they are not, test test fails (exit code 1).

Reply:
Since a diff of 2 different files doesn't through an error, I'm pretty sure it is not supposed to make the test fail. I tried it locally and when running diff of two different files in pytest, my test passed. 
Also, `diff` is Linux specific, if we start testing in MacOS or Windows the test will fail. Using an assert on the two df or on a numpy conversion of the df would be simpler (no need to start subprocesses) and will trigger if there is a difference.

Reply:
Oh ok, good to know (diff worked as I was expecting on my linux station). I'll check how I can use assert, thanks!

Reply:
or there is that (also it's extra dependency): https://pypi.org/project/csv-diff/

Reply:
Oh actually you're right it throws an error, my bad! But I still think it would be better to use assert when comparing both files (because it doesn't require a subprocess, it's not os dependant) I found this: https://stackoverflow.com/questions/45195388/best-way-to-compare-pandas-dataframe-with-csv-file to force datatypes, then you can use `assert df1.equals(df2)`

Reply:
I tested both `pandas.DataFrame.equals` and `pandas.testing.assert_frame_equal`. Even when forcing datatypes when reading the csv, some comparison still fail (seems related to a precision in float data). I tried all sorts of conversion and "almost equal" type of thing but was unable to make it work on all fields of the dataframes. I finally decided to use csv-diff as Julien suggested and added it to the requirements., it is much simpler and is working as expected.